### PR TITLE
Update release instructions

### PR DESCRIPTION
**Pull request by jasongrout**
Update release instructions

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
-
-To create a release, run the following:
+To create a release, update the version number in `nbclassic/__version__.py`, then run the following:
 
 ```
-git clean -dfx
+git clean -dfxx
```
Comment (afshin) [inline RELEASE.md:4]: Is one `x` superfluous?

Comment (jasongrout) [reply RELEASE.md:4]: No, it's extra assurance that *everything* is removed.

Comment (jasongrout) [reply RELEASE.md:4]: Wait, a while ago I looked up the options and that extra x was necessary. I can't find it in the manual now.

Comment (jasongrout) [reply RELEASE.md:4]: Ah, it's `-dffx`, not `-dfxx`.

**Pull request merged**